src/main/java/oss/fosslight/service/impl/HistoryServiceImpl.java

```diff
@@ -85,106 +85,121 @@ public Map<String, Object> getAsToBeHistoryDataByGrid(History history) {
 			HashMap<String, Object> sTblMap = new HashMap<String, Object>();
-			
+
 			// colNames 생성
-			List<Map<String, Object>> colNames = new ArrayList<Map<String, Object>>(); 
+			List<Map<String, Object>> colNames = new ArrayList<Map<String, Object>>();
 			// Param : String CD_NO
-	        // Result : Vector&String [ String CD_DTL_NO, String CD_SUB_NO, String CD_DTL_NM, String CD_DTL_NM, String CD_DTL_EXP, String CD_ORDER ]
+			// Result : Vector&String [ String CD_DTL_NO, String CD_SUB_NO, String CD_DTL_NM, String CD_DTL_NM, String CD_DTL_EXP, String CD_ORDER ]
 			HashMap<String, Object> colInfo =  new HashMap<String, Object>();
 			colInfo.put("key", "no");
 			colInfo.put("name", "no");
 			colInfo.put("order", "0");
 			colNames.add(colInfo);
-			
+
 			for(String[] v : CoCodeManager.getAllValues(sCdNm)){
-				 colInfo =  new HashMap<String, Object>();
-				 colInfo.put("key", v[2]);
-				 colInfo.put("name", v[4].split("\\|")[1]);
-				 colInfo.put("order", v[5]);
-				 
-				 colNames.add(colInfo);
+				colInfo =  new HashMap<String, Object>();
+				colInfo.put("key", v[2]);
+				colInfo.put("name", v[4].split("\\|")[1]);
+				colInfo.put("order", v[5]);
+
+				colNames.add(colInfo);
 			}
-			
-			sTblMap.put("type", inf[1]);		// model name
-			sTblMap.put("colNames", colNames);	// list columns
-			
+
+			sTblMap.put("type", inf[1]);      // model name
+			sTblMap.put("colNames", colNames); // list columns
+
 			// colModel 생성
 			List<Map<String, Object>> subList = new ArrayList<Map<String, Object>>();
-			
+
 			if(dMap != null && dMap.get(dtlCd[1]) != null){
 				// sub list 생성
 				int sCnt = 1;
-				
+
 				for(Map<String, Object> sMap : (List<Map<String, Object>>) dMap.get(dtlCd[1])){
 					Map<String, Object> sDataMap = new HashMap<String, Object>();
 					sDataMap.put("no", sCnt++);
-					
+
 					// row data 생성
 					for(String[] sDtlCd : CoCodeManager.getValues(sCdNm)){
-						String[] sInf = CoCodeManager.getCodeExpString(sCdNm, sDtlCd[0]).split("\\|");	// EXP : TYPE(String, Code, Array, Object) | NAME | CD_NO
-						
+						String[] sInf = CoCodeManager.getCodeExpString(sCdNm, sDtlCd[0]).split("\\|"); // EXP : TYPE(String, Code, Array, Object) | NAME | CD_NO
+
 						if(sInf[0].equals("String")){
 							sDataMap.put(sDtlCd[1], escapeSql(nvl((String)sMap.get(sDtlCd[1]))) );
 						}else if(sInf[0].equals("Code")){
 							sDataMap.put(sDtlCd[1], CoCodeManager.getCodeString(sInf[1], (String)sMap.get(sDtlCd[1])) );
-						}	
+						}
 					}
-					
+
 					subList.add(sDataMap);
 				}
 			}
-			
+
 			sTblMap.put("subList", subList);
-			
+
 			ret = sTblMap;
 		}
-		
+
 		return ret;
 	}
-		
+
 	public String escapeSql(String str){
 		return str == null ? "null" : String.format("%s", str.replace("'", "\\'").replace("\n", "\\n").replace("\r", "\\r").replace("\"", "\\\""));
 	}
+
+	// "Code" key 값이 Obligation 아닌 경우, 결과값 반환 함수
+	public String generateGeneralCodeRetValue(String[] inf, String[] dtlCd, Map<String, Object> dMap){

```

Comment:
Let's do not create a separate functions, `generateGeneralCodeRetValue` and `generateObligationCodeRetValue,` and put it directly inside the if statement if there is no need to be called anywhere else.